lib/matplotlib/colorbar.py

```diff
@@ -606,17 +606,19 @@ def _update_dividers(self):
         self.dividers.set_segments(segments)
 
     def _add_solids_patches(self, X, Y, C, mappable):
-        hatches = mappable.hatches * len(C)  # Have enough hatches.
+        hatches = mappable.hatches * (len(C) + 1)  # Have enough hatches.
         patches = []
+        hatch_offset = 1 if self._extend_lower() else 0
```

Comment:
Thinking about it a bit more, you could do:
```suggestion
        hatches = mappable.hatches * (len(C) + 1)  # Have enough hatches.
        if self._extend_lower():
            hatches = hatches[1:]
        patches = []
```
and then not need to change the indexing below.

Reply:
Yes, better idea!